### PR TITLE
Plated

### LINKED ISSUE #288: Plated optic
I'd like to see the [Plated type-class](https://hackage.haskell.org/package/lens-4.13/docs/Control-Lens-Plated.html#g:1) from Haskell lens implemented in Monocle:

```scala
trait Plated[A] {
  def plate: Traversal[A, A]
}

object Plated {
  def plate[A: Plated]: Traversal[A, A] = implicitly[Plated[A]].plate
  def children[A: Plated](a: A): List[A] = plate[A].getAll(a)
  def universe[A: Plated](a: A): Stream[A] = {
    val fold = plate[A].asFold
    def go(b: A): Stream[A] = fold.foldMap[Stream[A]](c => c #:: go(c))(b)
    go(a)
  }
  def transform[A: Plated](f: A => A)(a: A): A = plate[A].modify(f)(a)
}
```

The above might be a good start but I'd love to see functions like `rewrite`.

Some questions have to be answered, though, for example: Stream makes sense for `universe` but it might also make sense for `children` - what should we do?

### PR DESCRIPTION
**Pull request by puffnfresh**
To fix #288

### CODE REVIEW

File: core/src/main/scala/monocle/Plated.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          as => Applicative[F].map(Traverse[S].traverse(as)(f)) {
+            // Free.roll does not exist in 7.1
+            Free.liftF(_).flatMap(identity)
+          },
+          x => Applicative[F].point(Free.point(x))
+        )
+    }
+  }
+
+  implicit def listPlated[A]: Plated[List[A]] = new Plated[List[A]] {
```
Comment (julien-truffaut) [inline core/src/main/scala/monocle/Plated.scala]: would you mind to move the instances of `Plated` to `monocle.std.x`, in that case `monocle.std.list`

**Comment by puffnfresh** [comment]
Should `monocle.Plated` become `monocle.function.Plated` like `Each` is?

**Comment by julien-truffaut** [comment]
yes please, that's where we put all the typeclasses even though it is a poor choice of package name

**Comment by puffnfresh** [comment]
I've moved Plated to the right place now.

**Comment by julien-truffaut** [comment]
great! If you have time, would you mind to add a test or an example for `rewriteOf `, `transform`, `transformOf`

**Comment by puffnfresh** [comment]
@julien-truffaut I've added some specs for Plated, they're not brilliant but I think they're useful.

**Comment by julien-truffaut** [comment]
They are definitely useful. Thanks!

:+1:

**Pull request merged**